### DISCUSSION TITLE
What is the point/Why does godot have its own shader language

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#9283] [General Discussions]
its almost the same as glsl, with a few minor things changed from what i saw.  Why does this exist? what is the point ?

**Comment by Calinou** [comment]
The reason why the Godot shader language exists is to work as an abstraction layer over bare GLSL. This is done in the interest of usability, particularly for people who are new to shader programming. For instance, GLSL does not have a way to specify property hints for uniforms (so that the inspector can have range hints for uniforms) and there is no way to toggle parts of the shader based on which variables are being written to (e.g. transparency being toggled when `ALPHA` is written to).

The syntax is intentionally designed so that it's as similar as possible to GLSL, which makes it possible to copy-paste GLSL code and have it work out of the box in a lot of situations.

You **can** use GLSL in Godot, but it's a lot more complex and is only warranted for use cases that are currently not covered by the Godot shader language, such as [compute shaders](https://docs.godotengine.org/en/stable/tutorials/shaders/compute_shaders.html).

Also, this repository is for feature proposals. For general discussions and questions, please use the [forum](https://forum.godotengine.org/).
